src/components/MultiHopTrade/components/TradeAssetInput.tsx

```diff
@@ -46,7 +46,22 @@ const AssetInputWithAsset: React.FC<AssetInputLoadedProps> = props => {
   )
   const fiatBalance = bnOrZero(balance).times(marketData.price).toString()
 
-  return <TradeAmountInput balance={balance} fiatBalance={fiatBalance} {...props} />
+  const onMaxClick = useCallback(
+    async (isFiat: boolean) => {
+      const value = isFiat ? fiatBalance : balance
+      if (props.onChange) await props.onChange(value, isFiat)
+    },
```

Comment:
if we want to fix the ts warning about awaiting a non promise
```suggestion
    (isFiat: boolean) => {
      const value = isFiat ? fiatBalance : balance
      if (props.onChange) props.onChange(value, isFiat)
      return Promise.resolve()
    },
```

Reply:
Might as well too 🙏🏽 much nicer that way, ty! a322164e7ac01d4fa965c5ee717197dfb27649